src/os_posix/os_time.c

```diff
@@ -12,26 +12,26 @@
  * __wt_epoch --
  *	Return the time since the Epoch.
  */
-int
+void
 __wt_epoch(WT_SESSION_IMPL *session, struct timespec *tsp)
 {
 	WT_DECL_RET;
 
 #if defined(HAVE_CLOCK_GETTIME)
 	WT_SYSCALL(clock_gettime(CLOCK_REALTIME, tsp), ret);
 	if (ret == 0)
-		return (0);
-	WT_RET_MSG(session, ret, "clock_gettime");
+		return;
+	WT_PANIC_MSG(session, ret, "clock_gettime");
```

Comment:
@agorrod, it's not that time wouldn't move forward, it's that time would be zero'd out, so the callers would have to deal with:

1. a valid time followed by a zero'd out time,
2. a zero'd out time followed by a valid time, or
3. one or two zero'd out times (there are consumers of `__wt_epoch()` that use the time for purposes other than start/stop computations, for example, statistics logging file names).

We could handle number 1 by saving a copy of the first valid time somewhere (the `WT_SESSION_IMPL` structure?), but I don't see any way to handle the other two without significantly complicating the `__wt_epoch()` API.

Obviously, I don't think the change is risky or I wouldn't have suggested it (we've never seen this API fail, and the Windows implementation can't fail). But, if consensus is there's too much risk, then I think we have to kill the branch, I don't see a path to minimizing the risk that's less complicated than the existing error handling.